pkg/c3/util.c

```diff
@@ -0,0 +1,42 @@
+#include "util.h"
+
+c3_w
+_c3_printcap_mem_w(FILE *fil_u, c3_w wor_w, const c3_c *cap_c)
+{
+  return _c3_printcap_mem_z(fil_u, (c3_z)wor_w << 2, cap_c) >> 2;
+}
+
+c3_z
+_c3_printcap_mem_z(FILE *fil_u, c3_z byt_z, const c3_c *cap_c)
+{
+  c3_assert( 0 != fil_u );      /* ;;: I assume this is important from commit
+                                     f975ca908b143fb76c104ecc32cb59317ea5b198:
+                                     threads output file pointer through memory
+                                     marking and printing.
+
+                                    If not necessary, we can get rid of
+                                    c3_maid_w */
```

Comment:
The interface is even simpler if we can get rid of the null file assert in `_c3_printcap_mem_z`. I think we might be able to, but not sure. See inline comment.

If we can get rid of this, all calls to `c3_maid_w` can just call `c3_print_mem_w`.

Reply:
So the question is... when do we want a call to `c3_print_mem_w` to crash the process when the output file is null (rather than ignore the print). I would guess never